main.go

```diff
@@ -387,6 +388,28 @@ func main() {
 		p, err = plural.NewPluralProvider(cfg.PluralCluster, cfg.PluralProvider)
 	case "tencentcloud":
 		p, err = tencentcloud.NewTencentCloudProvider(domainFilter, zoneIDFilter, cfg.TencentCloudConfigFile, cfg.TencentCloudZoneType, cfg.DryRun)
+	case "webhook":
+		startedChan := make(chan struct{})
+		if cfg.RunAWSProviderAsWebhook {
```

Comment:
I suspect we'll want to move to a `--webhook` mechanism, where the binary only runs as the webhook. That would be helpful for building out the Helm chart support. I don't think this blocks this PR.